src/sdk/pynni/nni/protocol.py

```diff
@@ -43,9 +47,16 @@ def send(command, data):
     try:
         _lock.acquire()
         data = data.encode('utf8')
-        assert len(data) < 1000000, 'Command too long'
-        msg = b'%b%06d%b' % (command.value, len(data), data)
-        logging.getLogger(__name__).debug('Sending command, data: [%s]', msg)
+        logging.getLogger(__name__).debug('Sending command %s, data: [%s]', command.value, data)
+        if len(data) <= _max_msg_len:
+            msg = b'%b%06d%b' % (command.value, len(data), data)
+        else:
+            msg = b'%b______%b' % (command.value, data[:_max_msg_len])
+            data = data[_max_msg_len:]
+            while len(data) > _max_msg_len:
+                msg += b'_C999999%b' % data[:_max_msg_len]
+                data = data[_max_msg_len:]
+            msg += b'_E%06d%b' % (len(data), data)
         _out_file.write(msg)
```

Comment:
if we can write data with any large content length, why not simply increase the number of bits for representing data length?